model/lifecycle.go

```diff
@@ -338,13 +338,30 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
 			return errors.WithStack(err)
 		}
 	}
+	allFinishedTasks, err := getTasksToReset(taskIds)
+	if err != nil {
+		return errors.Wrap(err, "getting finished tasks")
+	}
+	if len(allFinishedTasks) == 0 {
+		return nil
+	}
+	err = restartTasks(allFinishedTasks, caller)
+	if err != nil {
+		return errors.Wrap(err, "restarting tasks")
+	}
+	return errors.Wrap(setVersionStatus(versionId, evergreen.VersionStarted), "changing version status")
+}
+
+// getTasksToReset returns all finished tasks that should be reset given an initial input list of
+// finished taskIds.
+func getTasksToReset(taskIds []string) ([]task.Task, error) {
 	finishedTasks, err := task.FindAll(db.Query(task.ByIdsAndStatus(taskIds, evergreen.TaskCompletedStatuses)))
 	if err != nil {
-		return errors.WithStack(err)
+		return nil, errors.WithStack(err)
 	}
 	allFinishedTasks, err := task.AddParentDisplayTasks(finishedTasks)
 	if err != nil {
-		return errors.WithStack(err)
+		return nil, errors.WithStack(err)
 	}
 	// Remove execution tasks in case the caller passed both display and execution tasks.
 	// The functions below are expected to work if just the display task is passed.
```

Comment:
Functions below isn't accurate anymore; maybe the function comment should say that it removes execution tasks?

Reply:
Sounds good, moved to function level comment